src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -83,9 +88,10 @@ abstract class Kernel implements KernelInterface, RebootableInterface, Terminabl
     const END_OF_MAINTENANCE = '07/2021';
     const END_OF_LIFE = '07/2021';
 
-    public function __construct(string $environment, bool $debug)
+    public function __construct(string $mode, bool $debug)
     {
-        $this->environment = $environment;
+        $this->mode = $mode;
+        $this->environment = $mode;
```

Comment:
this is an incomplete BC layer. As the properties are protected, it misses handling property writes done in child classes.